### PR TITLE
feat/subflow

### PR DESCRIPTION
**Pull request by brian-mulier-p**
closes kestra-io/kestra#774

### CODE REVIEW

File: src/utils/Utils.js [hunk 1]
```diff
@@ -44,4 +45,8 @@ export default class Utils {
 
         return humanizeDuration(value * 1000, options).replace(/\.([0-9])s$/i, ".$10s")
     }
+
+    static afterLastDot(str) {
+        return str.split(".").pop() ?? str;
```
Comment (Skraye) [inline src/utils/Utils.js:50]:
```suggestion
        return str.split(".").pop();
```

File: src/components/nodes/BasicNode.vue [hunk 2]
```diff
@@ -178,16 +178,27 @@
...
+
+                if (!this.data.node?.task) {
+                    return undefined;
+                }
+
+                return this.data.node.task.type;
             },
-            cls() {
-                return this.data.node?.task ? this.data.node.task.type : this.data.node?.trigger ? this.data.node.trigger.type : null
+            hoverTooltip() {
```
Comment (Skraye) [inline src/components/nodes/BasicNode.vue:192]: No need to display the whole path, only the ID, way too much useless information however (the point of the tooltip is to display ID too long to fit in a node)
image

File: src/components/nodes/ClusterNode.vue [hunk 3]
```diff
@@ -37,6 +54,16 @@
                 tooltips: [],
             }
         },
+        computed: {
+            clusterName() {
+                const taskNode = this.data.taskNode;
+                if (taskNode?.type?.endsWith("SubflowGraphTask")) {
+                    return taskNode.task.namespace + " " + taskNode.task.flowId;
```
[the remark this answers is not in the document]
Comment (brian-mulier-p) [reply src/components/nodes/ClusterNode.vue:61]: I asked @tchiotludo  when I did this and he told me to use space to know that it includes the flowId

File: src/components/nodes/BasicNode.vue (hunk 2 above)
Comment (brian-mulier-p) [reply src/components/nodes/BasicNode.vue:192]: I like the idea of easily understanding how nested we are with the hover but okay

**Comment by Skraye** [comment]
Subflow tasks should be greyed out when dragging
image

**Comment by Skraye** [comment]
Flowable tasks should be swappable with same level flowable and their children (here, I am trying to swap a switch with is `errors` task list)
image

**Comment by brian-mulier-p** [comment]
I made the changes :+1:

**Review by Skraye — APPROVED** [review]
LGTM

**Pull request merged**